File: v1/src/main/java/com/api/v1/medical_appointments/responses/CompletedMedicalAppointmentResponseDto.java
package com.api.v1.medical_appointments.responses;

import com.api.v1.customers.responses.CustomerResponseDto;
import com.api.v1.doctors.responses.DoctorResponseDto;
import com.api.v1.medical_appointments.domain.exposed.MedicalAppointment;
import com.api.v1.medical_appointments.enums.MedicalAppointmentStatus;

import java.time.LocalDateTime;

public final class CompletedMedicalAppointmentResponseDto extends MedicalAppointmentResponseDto {

    private LocalDateTime completedAt;

    private CompletedMedicalAppointmentResponseDto() {}

    private CompletedMedicalAppointmentResponseDto(String id,
                                                  CustomerResponseDto customer,
                                                  DoctorResponseDto doctor,
                                                  MedicalAppointmentStatus status,
                                                  LocalDateTime bookedAt,
                                                  LocalDateTime completedAt
    ) {
        super(id, customer, doctor, status, bookedAt);
        this.completedAt = completedAt;
    }

    public static CompletedMedicalAppointmentResponseDto from(MedicalAppointment medicalAppointment) {
        return new CompletedMedicalAppointmentResponseDto(
                medicalAppointment.getId(),
                medicalAppointment.getCustomer().toDto(),
                medicalAppointment.getDoctor().toDto(),
                medicalAppointment.getStatus(),
                medicalAppointment.getBookedAt(),
                medicalAppointment.getCompletedAt()
        );
    }

    public LocalDateTime getCompletedAt() {
        return completedAt;
    }
}

File: v2/src/main/kotlin/com/api/v2/medical_appointment/CompletedMedicalAppointmentResponseDTO.kt
package com.api.v2.medical_appointment

import com.api.v2.customers.CustomerResponseDTO
import com.api.v2.doctors.DoctorResponseDTO
import com.api.v2.medical_appointment.enums.MedicalAppointmentStatus
import java.time.LocalDateTime

class CompletedMedicalAppointmentResponseDTO(
    id: String,
    customer: CustomerResponseDTO,
    doctor: DoctorResponseDTO,
    status: MedicalAppointmentStatus,
    bookedAt: LocalDateTime,
    createdAt: LocalDateTime,
    completedAt: LocalDateTime
): MedicalAppointmentResponseDTO(id, customer, doctor, status,  bookedAt, createdAt)